edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -169,6 +170,8 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                         IotHubConnectionStringBuilder connectionStringParser = IotHubConnectionStringBuilder.Create(deviceConnectionString);
                         deviceId = connectionStringParser.DeviceId;
                         iothubHostname = connectionStringParser.HostName;
+                        moduleId = connectionStringParser.ModuleId;
+                        PersistentStorageValidation.ValidateStorageIdentity(storagePath, deviceId, iothubHostname, moduleId, Option.None<string>(), logger);
```

Comment:
Let's put this behind a flag.. say something like "CleanupStateOnReset". 
Default should be true. If it is false, then we don't do this.